### DISCUSSION TITLE
False positive results with imports via `paths` alias

### DISCUSSION THREAD

**Discussion by jtbandes** [nadeesha/ts-prune#129] [Q&A]
Hello @nadeesha! I just tried using ts-prune for the first time on [this repository](https://github.com/foxglove/studio) and I'm confused by the results, because many/most of them seem to be false positives.

The repo is a monorepo with several packages. This is the command I ran to try out ts-prune on the biggest package ([here's the tsconfig.json](https://github.com/foxglove/studio/blob/7142d5e1c810cfc427fe2d2a6a860ed81424f132/packages/studio-base/tsconfig.json)):

```
yarn dlx ts-prune -p packages/studio-base/tsconfig.json
```

The results look like this:
```
packages/studio-base/src/App.tsx:31 - default
packages/studio-base/src/AppSetting.ts:5 - AppSetting
packages/studio-base/src/index.ts:11 - App
packages/studio-base/src/index.ts:12 - NetworkInterface
packages/studio-base/src/index.ts:12 - OsContext
....a bunch more from index.ts....
packages/studio-base/src/OsContextSingleton.ts:13 - default
packages/studio-base/src/Workspace.tsx:142 - default
....lots more results....
```

For example, the first result is the [default export of `App.tsx`](https://github.com/foxglove/studio/blob/7142d5e1c810cfc427fe2d2a6a860ed81424f132/packages/studio-base/src/App.tsx#L31). This is [re-exported from `index.tsx`](https://github.com/foxglove/studio/blob/7142d5e1c810cfc427fe2d2a6a860ed81424f132/packages/studio-base/src/index.ts#L11), so I wouldn't expect it to be considered unused.

For another example let's look at the [default export from `OsContextSingleton.ts`](https://github.com/foxglove/studio/blob/7142d5e1c810cfc427fe2d2a6a860ed81424f132/packages/studio-base/src/OsContextSingleton.ts#L13) — this is imported as `import OsContextSingleton from "@foxglove/studio-base/OsContextSingleton";` in several places ([here's one example](https://github.com/foxglove/studio/blob/7142d5e1c810cfc427fe2d2a6a860ed81424f132/packages/studio-base/src/components/Preferences.tsx#L20)).

Is it our [`paths` setting](https://github.com/foxglove/studio/blob/7142d5e1c810cfc427fe2d2a6a860ed81424f132/packages/studio-base/tsconfig.json#L10) that's throwing off ts-prune? Or am I doing something else wrong?

Thanks for your help! This seems like a great project and it would be awesome if it worked for us.

**Comment by connorads** [comment]
Hey @jtbandes, I had the same experience. Did you manage to resolve it?

_PS. This is my first time using GitHub discussions. I guess it might be like StackOverflow and I'm not supposed to put questions here 😬_

**Reply by jtbandes** [reply]
Nope, I didn’t learn anything more after I posted this 😕

**Reply by connorads** [reply]
For anyone who discovers this thread: Conversation moved to issue https://github.com/nadeesha/ts-prune/issues/139
